dask/array/core.py

```diff
@@ -2825,37 +2829,53 @@ def choose(a, choices):
     return elemwise(variadic_choose, a, *choices)
 
 
-where_error_message = """
-The dask.array version of where only handles the three argument case.
+chunks_none_error_message = """
+You must specify a chunks= keyword argument.
+This specifies the chunksize of your array blocks.
 
-    da.where(x > 0, x, 0)
+See the following documentation page for details:
+  http://dask.pydata.org/en/latest/array-creation.html#chunks
+""".strip()
 
-and not the single argument case
 
-    da.where(x > 0)
+def isnonzero(a):
+    a = asarray(a)
+    if issubclass(a.dtype.type, Number):
+        return (a != 0)
+    if issubclass(a.dtype.type, (np.bytes_, np.unicode_)):
+        ######################################################
+        # Treats empty or whitespace only strings as False.  #
+        # This behavior mimics NumPy's own behavior.         #
+        #                                                    #
+        # xref: https://github.com/numpy/numpy/issues/9462   #
+        ######################################################
+        np_isnonzero = np.vectorize(lambda v: bool(v.strip()), otypes=[bool])
+        return a.map_blocks(np_isnonzero, dtype=bool)
+    else:
+        np_isnonzero = np.vectorize(lambda v: bool(v), otypes=[bool])
+        return a.map_blocks(np_isnonzero, dtype=bool)
```

Comment:
We could try to do `astype(bool)`. This seems to work for a couple`object` arrays I tried. Also works for numeric types. Not sure if there are any other cases that would cause us trouble.